pkgs/shells/nushell/default.nix

```diff
@@ -1,3 +1,9 @@
+with import <nixpkgs> {
+  overlays = [
+    (import (fetchTarball "https://github.com/oxalica/rust-overlay/archive/master.tar.gz"))
+  ];
+};
```

Comment:
This would break Nixpkgs in pure evaluation mode, due to using `<nixpkgs>` which uses the `NIX_PATH` environment variable and using `fetchTarball` without a hash (thus it is not locked to a specific file).

If you really wanted to do this, you could replace `<nixpkgs>` with `pkgs.path` and `fetchTarball` with `fetchFromGitHub`. It would also be a good idea to change:

```nix
with import <nixpkgs> {
  overlays = [ (...) ];
}
```

to:

```nix
with pkgs.extend {
  (...)
}
```

Reply:
Even using another way to fetch, it would still be an import from derivation and doesn't work if you evaluate it in read-only mode.

Reply:
I just removed that overlay part and locally tested the change based on staging (which contains a more recent cargo). It works fine. Now all we have to do is wait until 1.59 hits master (as already suggested above) :-)